homeassistant/components/ps4/media_player.py

```diff
@@ -375,7 +375,12 @@ def media_stop(self):
         self.send_remote_control('ps')
 
     def select_source(self, source):
-        """Select input source."""
+        """Select input source. If device is off,
+           turn it on and wait until it's on."""
+        if (self._state == STATE_OFF):
+            self.turn_on()
+            while (self._state == STATE_OFF):
+                pass
```

Comment:
This will eat all CPU until the state changes, that is not acceptable.

You will probably need a synchronization primitive such as `asyncio.Event` but that is a bit tricky because this integration is not async.

Reply:
Would a time.sleep(1) be acceptable?

Reply:
No, sleeping isn't fine either I'd say.

Reply:
We definitely can't use `time.sleep` in our thread pool. Agreed with @amelchio: you need some way for the underlying component to notify you that the system is on.

Reply:
Thanks for the feedback. I'll with withdraw this PR og look into solving this some other way.